Cstruct 1.6.0

dsheets opened the pull request:
Cstruct 1.6.0 contains compare function (added by @pqwy...) that causes this expression open to fail.

Also, I added a merlin file.

hannesm commented:
shouldn't this compare function than just go away, and the cstruct one be used!?

dsheets commented:
Indeed. I wasn't paying attention to the functionality and was simply looking to avoid the error.

I've removed `Asn_writer.cs_compare` now and updated the `opam` file. Let me know if you'd like the history hacked.

avsm commented:
incidentally, it would be useful if the `opam` file here could be bumped in `version` so that constraints work when pinning.

pqwy commented:
Thanks! But... it's incorrect now. :sob:  Following the [discussion](https://github.com/mirage/ocaml-cstruct/pull/45), the new `Cstruct.compare` implements an arbitrary total order, while this deleted function specifically does lexicographic comparison, as required by ASN.1/BER encoding of SETs.

So I'll close this one; In the interest of promptness, I'm re-pushing the quickfix for name-collision. @dsheets Feel free to re-PR `.merlin`.

@avsm Will revbump ASAP when I merge the other branch of this I have.

I think this covers all the points. *Looks around*

avsm commented:
Sounds good to me! Perhaps reflect the difference explicitly in the no crypto comments



> On 4 Apr 2015, at 03:51, David Kaloper <notifications@github.com> wrote:
> 
> Thanks! But... it's incorrect now. Following the discussion, the new Cstruct.compare implements an arbitrary total order, while this deleted function specifically does lexicographic comparison, as required by ASN.1/BER encoding of SETs.
> 
> So I'll close this one; In the interest of promptness, I'm re-pushing the quickfix for name-collision. @dsheets Feel free to re-PR .merlin.
> 
> @avsm Will revbump ASAP when I merge the other branch of this I have.
> 
> I think this covers all the points. Looks around
> 
> —
> Reply to this email directly or view it on GitHub.
>

dsheets commented:
b62306215e422e3ef8301d25130d79b7b35349f3 is exactly ade29dc. If you want the merlin file, feel free to cherry-pick the commit.
